orte/mca/oob/tcp/oob_tcp_listener.c

```diff
@@ -748,18 +741,18 @@ static void* listen_thread(opal_object_t *obj)
                                                 &addrlen);
                 if (pending_connection->fd < 0) {
                     OBJ_RELEASE(pending_connection);
-                    if (opal_socket_errno != EAGAIN || 
-                        opal_socket_errno != EWOULDBLOCK) {
-                        if (EMFILE == opal_socket_errno) {
-                            ORTE_ERROR_LOG(ORTE_ERR_SYS_LIMITS_SOCKETS);
-                            orte_show_help("help-orterun.txt", "orterun:sys-limit-sockets", true);
-                        } else {
-                            opal_output(0, "mca_oob_tcp_accept: accept() failed: %s (%d).",
-                                        strerror(opal_socket_errno), opal_socket_errno);
-                        }
+                    if (EAGAIN == opal_socket_errno ||
+                        EWOULDBLOCK == opal_socket_errno) {
+                        continue;
+                    } else if (EMFILE == opal_socket_errno) {
+                        ORTE_ERROR_LOG(ORTE_ERR_SYS_LIMITS_SOCKETS);
+                        orte_show_help("help-orterun.txt", "orterun:sys-limit-sockets", true);
+                        goto done;
+                    } else {
+                        opal_output(0, "mca_oob_tcp_accept: accept() failed: %s (%d).",
+                                    strerror(opal_socket_errno), opal_socket_errno);
                         goto done;
                     }
-                    continue;
```

Comment:
@jsquyres this seems like a change in behavior, yes? It now looks like any time we get an accept failure, and it isn't EAGAIN or EWOULDBLOCK, then we exit the listener thread. Prior to this change, we would continue to listen.

Is that what you intended? If so, why?